docker-java-transport-okhttp/src/main/java/com/github/dockerjava/okhttp/OkDockerHttpClient.java

```diff
@@ -0,0 +1,284 @@
+                    clientBuilder.socketFactory(new NamedPipeSocketFactory(socketPath));
+                }
+
+                clientBuilder
+                    .connectionPool(new ConnectionPool(0, 1, TimeUnit.SECONDS))
+                    .dns(hostname -> {
+                        if (hostname.endsWith(SOCKET_SUFFIX)) {
+                            return Collections.singletonList(InetAddress.getByAddress(hostname, new byte[]{0, 0, 0, 0}));
+                        } else {
+                            return Dns.SYSTEM.lookup(hostname);
+                        }
+                    });
+                break;
+            default:
+        }
+
+        boolean isSSL = false;
+        SSLConfig sslConfig = dockerClientConfig.getSSLConfig();
+        if (sslConfig != null) {
+            try {
+                SSLContext sslContext = sslConfig.getSSLContext();
+                if (sslContext != null) {
+                    isSSL = true;
+                    clientBuilder.sslSocketFactory(sslContext.getSocketFactory(), new TrustAllX509TrustManager());
+                }
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        client = clientBuilder.build();
+
+        streamingClient = client.newBuilder().build();
+
+        HttpUrl.Builder baseUrlBuilder;
+
+        switch (dockerHost.getScheme()) {
+            case "unix":
+            case "npipe":
+                baseUrlBuilder = new HttpUrl.Builder()
+                    .scheme("http")
+                    .host("docker" + SOCKET_SUFFIX);
+                break;
+            case "tcp":
+                baseUrlBuilder = new HttpUrl.Builder()
+                    .scheme(isSSL ? "https" : "http")
+                    .host(dockerHost.getHost())
+                    .port(dockerHost.getPort());
+                break;
+            default:
+                baseUrlBuilder = HttpUrl.get(dockerHost.toString()).newBuilder();
+        }
+        baseUrl = baseUrlBuilder.build();
+    }
+
+    private RequestBody toRequestBody(Request request) {
+        InputStream body = request.body();
+        if (body != null) {
+            return new RequestBody() {
+                @Override
+                public MediaType contentType() {
+                    return null;
+                }
+
+                @Override
+                public void writeTo(BufferedSink sink) throws IOException {
+                    sink.writeAll(Okio.source(body));
+                }
+            };
+        }
+        switch (request.method()) {
+            case "POST":
+                return RequestBody.create(null, "");
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public Response execute(Request request) {
+        String url = baseUrl.toString();
+        if (url.endsWith("/") && request.path().startsWith("/")) {
+            url = url.substring(0, url.length() - 1);
+        }
```

Comment:
`private String buildUrl(baseUrl, request)`

Reply:
it is not really `buildUrl`, which kinda suggests that extracting it to a method may not be as helpful :D